src/datamigration/azext_datamigration/manual/custom.py

```diff
@@ -0,0 +1,242 @@
+# Assessment helper function to return the default output folder path depending on OS environment.
+# -----------------------------------------------------------------------------------------------------------------
+def get_default_output_folder():
+
+    osPlatform = platform.system()
+
+    if osPlatform.__contains__('Linux'):
+        defaultOutputPath = os.path.join(os.getenv('USERPROFILE'), ".config", "Microsoft", "SqlAssessment")
+    elif osPlatform.__contains__('Darwin'):
+        defaultOutputPath = os.path.join(os.getenv('USERPROFILE'), "Library", "Application Support", "Microsoft", "SqlAssessment")
+    else:
+        defaultOutputPath = os.path.join(os.getenv('LOCALAPPDATA'), "Microsoft", "SqlAssessment")
+
+    return defaultOutputPath
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Register Sql Migration Service on IR command Implementation.
+# -----------------------------------------------------------------------------------------------------------------
+def datamigration_register_ir(auth_key,
+                              ir_path=None):
+
+    validate_os_env()
+
+    if not is_user_admin():
+        raise CLIError("Failed: You do not have Administrator rights to run this command. Please re-run this command as an Administrator!")
+    validate_input(auth_key)
+    if ir_path is not None:
+        if not os.path.exists(ir_path):
+            raise CLIError(f"Invalid Integration Runtime MSI path : {ir_path}. Please provide a valid Integration Runtime MSI path")
+        install_gateway(ir_path)
+
+    register_ir(auth_key)
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check whether the command is run as admin.
+# -----------------------------------------------------------------------------------------------------------------
+def is_user_admin():
+
+    try:
+        isAdmin = os.getuid() == 0
+    except AttributeError:
+        isAdmin = ctypes.windll.shell32.IsUserAnAdmin() != 0
+
+    return isAdmin
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to validate key input. More checks to be added here using regex.
+# -----------------------------------------------------------------------------------------------------------------
+def validate_input(key):
+    if key == "":
+        raise CLIError("Failed: IR Auth key is empty. Please provide a valid auth key.")
+
+
+# -----------------------------------------------------------------------------------------------------------------
+# Helper function to check whether SHIR is installed or not.
+# -----------------------------------------------------------------------------------------------------------------
+def check_whether_gateway_installed(name):
+
+    import winreg
```

Comment:
Do you still need to add winreg as python DEPENDENCIES?

Reply:
"winreg" is actually a default module available through python installation so we didn't need to add it in dependencies. But Linux python doesn't has this as a result the pipeline was giving an error. So I added this to specific functions which use this module as this command isn't supported in Linux yet.